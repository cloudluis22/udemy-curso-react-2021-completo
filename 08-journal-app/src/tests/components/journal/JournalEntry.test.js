import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

import { JournalEntry } from '../../../components/journal/JournalEntry';

/*jest.mock('../../../actions/auth', () => ({
    startLogout: jest.fn(),
}));*/

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: 10432413,
    date: 123231,
    title: 'hola',
    body: 'mundo',
    url: 'https://linkdetesteo.com/foto.jpg'
}
const wrapper = mount(

    <Provider store={store}>
        <JournalEntry {...note} />
    </Provider>
);


describe('Pruebas en el componente funcional de "<JournalEntry />".', () => {

    test('Debe de renderizarse correctamente y hacer match con el snapshot.', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de activar la nota al hacer click.', () => {

        wrapper.find('.journal__entry').simulate('click');

        expect(store.dispatch).toHaveBeenCalled();

    });

});