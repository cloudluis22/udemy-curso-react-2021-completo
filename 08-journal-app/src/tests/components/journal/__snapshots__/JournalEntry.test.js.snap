// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente funcional de "<JournalEntry />". Debe de renderizarse correctamente y hacer match con el snapshot. 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="journal__entry pointer"
  >
    <div
      className="journal__entry-picture"
      style={
        Object {
          "backgroundImage": "url(https://linkdetesteo.com/foto.jpg)",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
         
        hola
         
      </p>
      <p
        className="journal__entry-content"
      >
         
        mundo
         
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
         
        Wednesday
         
      </span>
      <h4>
         
        31st
         
      </h4>
    </div>
  </div>
</div>
`;
